### PR TITLE
test(dashboard): add StatusBar and Toast component test coverage

### LINKED ISSUE #1717: test(dashboard): add StatusBar/FooterBar test coverage
## Goal
StatusBar and FooterBar components lack test coverage.

## Acceptance Criteria
- [ ] Tests for StatusBar rendering with different session states
- [ ] Tests for FooterBar with different connection phases
- [ ] Tests for model/permission mode display
- [ ] Tests for cost display formatting
- [ ] TDD: RED → GREEN → REFACTOR

### LINKED ISSUE #1723: test(dashboard): add Toast component tests
## Goal
Add unit tests for the Toast/notification component.

## Acceptance Criteria
- [ ] Tests for rendering different notification types
- [ ] Tests for auto-dismiss timing
- [ ] Tests for manual dismiss
- [ ] TDD: RED → GREEN → REFACTOR

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

Adds test files for two components that had zero coverage:

**StatusBar.test.tsx** (12 tests):
- Renders with testid, shows model/cost/context conditionally
- Busy indicator shown/hidden based on `isBusy`
- Agent badge with correct singular/plural text, hidden when 0 or absent

**Toast.test.tsx** (8 tests):
- Container rendering with `aria-live="assertive"`
- All toast messages rendered
- Close button calls `onDismiss` with correct id
- Empty state renders no toasts
- 5s auto-dismiss via fake timers
- No dismiss before 5s (4999ms check)

## Test plan

- [x] 12 StatusBar tests pass
- [x] 8 Toast tests pass
- [x] 20/20 total

Closes #1717
Closes #1723

### CODE REVIEW

**Comment by blamechris** [comment]
## Code Review: PR #1762

### Summary

Adds unit test files for two previously uncovered dashboard components — `StatusBar` and `Toast`. The tests are focused, well-structured, and consistent with the patterns used throughout the rest of the dashboard test suite. This is a pure test addition (137 lines, 0 deletions) that closes two open issues.

### Strengths

- Tests are isolated and independent — no shared state bleeds between cases
- `afterEach(cleanup)` used correctly (consistent with every other test file in the directory)
- Toast timer tests correctly call `vi.useRealTimers()` after each fake-timer block, preventing global test pollution
- `makeItems()` helper keeps fixture construction concise without over-abstracting
- Auto-dismiss timing test covers both the positive case (5000ms fires) and the boundary negative case (4999ms does not), which is exactly the right level of rigor
- Accessibility attribute (`aria-live=\"assertive\"`) is explicitly verified — good practice for notification components
- `role=\"alert\"` on individual toast items is rendered in the component — not tested here, but not strictly necessary for this PR scope

### Issues Found

#### Critical (Must Fix)

None.

#### Suggestions (Should Consider)

| File | Line | Suggestion | Rationale |
|------|------|------------|-----------|
| `StatusBar.test.tsx` | — | Add a test for `cost={0}` | The component renders `$0.0000` when `cost` is `0` (uses `cost != null` guard, not `cost`). A zero-cost render is a valid real-world state (e.g. fresh session). Without a test, a future refactor to `cost &&` would silently regress. |
| `Toast.test.tsx` | — | Consider verifying `onDismiss` call count on auto-dismiss | `toHaveBeenCalledWith('toast-0')` passes even if it was called multiple times. `toHaveBeenCalledOnce()` + `.toHaveBeenCalledWith` together gives tighter timer cleanup guarantees. Low priority. |

#### Nitpicks

- The `describe` blocks don't have a nested structure (all `it()` calls are flat) — this is consistent with simpler components in the codebase, so it's appropriate here.
- Issue #1717's title says "StatusBar/FooterBar" but FooterBar already has coverage from #1504. The PR correctly closes #1717 since the StatusBar portion satisfies the acceptance criteria, and FooterBar was never missing coverage at merge time.

### Deferred Items (Follow-Up Issues)

| Suggestion | Issue | Rationale for deferral |
|------------|-------|------------------------|
| `cost={0}` edge case test for StatusBar | [#1764](https://github.com/blamechris/chroxy/issues/1764) | Minor omission; doesn't block this PR. Tracked for the next test-coverage pass. |

### Architecture Notes

These tests live in the correct location (`dashboard-next/src/components/`) alongside their implementation files, consistent with the entire suite. The vitest + jsdom + `@testing-library/react` stack is correct. No new dependencies introduced.

### Verdict

- [x] **Approve** — Ready to merge. Both target components are well-covered. The `cost={0}` gap is a minor edge case tracked in a follow-up issue, not a blocker.

File: packages/server/src/dashboard-next/src/components/StatusBar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  })
+
+  it('shows model name when provided', () => {
+    render(<StatusBar model="claude-sonnet" />)
+    expect(screen.getByText('claude-sonnet')).toBeInTheDocument()
+  })
+
+  it('does not show model when not provided', () => {
+    render(<StatusBar />)
+    expect(screen.queryByText(/claude/)).not.toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/src/dashboard-next/src/components/StatusBar.test.tsx]: **FIX**

Fixed in `b69893b0`

**Change:** Replaced the weak `/claude/` regex with a CSS class query. Note: the component does not have `data-testid='status-model'` (Copilot's suggestion would have failed), but `.status-model` className is present and sufficient.

```diff
- it('does not show model when not provided', () => {
-   render(<StatusBar />)
-   expect(screen.queryByText(/claude/)).not.toBeInTheDocument()
- })
+ it('does not show model element when not provided', () => {
+   const { container } = render(<StatusBar />)
+   expect(container.querySelector('.status-model')).toBeNull()
+ })
```

File: packages/server/src/dashboard-next/src/components/Toast.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { describe, it, expect, vi, afterEach } from 'vitest'
+import { render, screen, fireEvent, cleanup, act } from '@testing-library/react'
+import { Toast, type ToastItem } from './Toast'
+
+afterEach(cleanup)
+
+const makeItems = (...messages: string[]): ToastItem[] =>
+  messages.map((m, i) => ({ id: `toast-${i}`, message: m }))
+
+describe('Toast', () => {
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/src/dashboard-next/src/components/Toast.test.tsx]: **FIX**

Fixed in `b69893b0`

**Change:** Removed the `Toast` describe block from `Modal.test.tsx` and made `Toast.test.tsx` the canonical home. The Modal test file header was updated to reflect this. All tests from the Modal suite (including the timer-cleanup and `role="alert"` cases) were merged into `Toast.test.tsx`.

```diff
- // Modal, CreateSessionModal, Toast tests (#1164)
+ // Modal and CreateSessionModal tests (#1164)
+ // Toast tests are in Toast.test.tsx
```

File: packages/server/src/dashboard-next/src/components/Toast.test.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  })
+
+  it('does not auto-dismiss before 5 seconds', async () => {
+    vi.useFakeTimers()
+    const onDismiss = vi.fn()
+    render(<Toast items={makeItems('Not yet')} onDismiss={onDismiss} />)
+    await act(async () => { vi.advanceTimersByTime(4999) })
+    expect(onDismiss).not.toHaveBeenCalled()
+    vi.useRealTimers()
+  })
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/src/dashboard-next/src/components/Toast.test.tsx]: **FIX**

Fixed in `b69893b0`

**Change:** Extracted timer tests into a nested `describe('timer behaviour')` block with `beforeEach`/`afterEach` hooks that guarantee `vi.useRealTimers()` runs even on assertion failures.

```diff
- it('auto-dismisses after 5 seconds', async () => {
-   vi.useFakeTimers()
-   ...
-   vi.useRealTimers()
- })
-
- it('does not auto-dismiss before 5 seconds', async () => {
-   vi.useFakeTimers()
-   ...
-   vi.useRealTimers()
- })
+ describe('timer behaviour', () => {
+   beforeEach(() => { vi.useFakeTimers() })
+   afterEach(() => { vi.runOnlyPendingTimers(); vi.useRealTimers() })
+
+   it('auto-dismisses after 5 seconds', async () => { ... })
+   it('does not auto-dismiss before 5 seconds', async () => { ... })
+   it('clears auto-dismiss timer when manually closed', async () => { ... })
+ })
```
